homeassistant/components/deconz.py

```diff
@@ -0,0 +1,198 @@
+"""
+Support for deCONZ support.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/deconz/
+"""
+
+import asyncio
+import os
+import logging
+import voluptuous as vol
+
+from homeassistant.config import load_yaml_config_file
+from homeassistant.const import (
+    CONF_API_KEY, CONF_EVENT, CONF_HOST, CONF_ID, CONF_PASSWORD, CONF_PORT,
+    CONF_USERNAME, EVENT_HOMEASSISTANT_STOP)
+from homeassistant.core import callback, EventOrigin
+from homeassistant.helpers import config_validation as cv
+from homeassistant.helpers import discovery
+from homeassistant.util import slugify
+from homeassistant.util.json import load_json, save_json
+
+REQUIREMENTS = ['pydeconz==20']
+
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = 'deconz'
+
+DECONZ_DATA = 'deconz_data'
+CONFIG_FILE = 'deconz.conf'
+
+CONF_SWITCH_AS_EVENT = 'switch_as_event'
```

Comment:
This should not be configurable. 

Buttons can be events, not sure how a switch can be? Switches are on or off and can be toggled.

Reply:
Switches in this context are remotes like the Hue Dimmer. We had a discussion about this earlier in Discord